### PR TITLE
Add generate breakpoints

### PR DESCRIPTION
**Pull request by kllevin**

### CODE REVIEW

File: utilities/_u-text.scss [hunk 1]
```diff
@@ -19,36 +27,55 @@
 %u-text-size-small,
 .u-text-size-small {@include font-size($font-size-small);}
 
+@include generate-at-breakpoints('.u-table', $text-breakpoints) {@include font-size($font-size-small);}
```
Comment (AlecRaeside) [inline utilities/_u-text.scss]: Needs to wrap at 80 chars

File: utilities/_u-momentum-scrolling.scss [hunk 2]
```diff
@@ -15,8 +15,22 @@
...
 %u-momentum-scrolling,
 .u-momentum-scrolling {
   overflow: auto;
   -webkit-overflow-scrolling: touch;
+}
+
+@include generate-at-breakpoints('.u-momentum-scrolling',
+$momentum-scrolling-breakpoints) {
+ overflow: auto;
+  -webkit-overflow-scrolling: touch;
```
Comment (chris-pearce) [inline utilities/_u-momentum-scrolling.scss]: Blank space.

Comment (chris-pearce) [inline utilities/_u-momentum-scrolling.scss]: Thinking about this more I think we don't need it for this partial.

File: utilities/_u-table.scss [hunk 3]
```diff
@@ -61,9 +77,14 @@ $table-striped-background-cell-colour:  $table-border-colour !default;
...
 .u-table--striped {
-  tbody tr:nth-of-type(odd) td {background-color: $table-striped-background-cell-colour;}
+  tbody tr:nth-of-type(odd) td {background-color:
+    $table-striped-background-cell-colour;}
 }
 
+@include generate-at-breakpoints('.u-table--striped', $table-breakpoints) {
+  tbody tr:nth-of-type(odd) td {background-color:
+    $table-striped-background-cell-colour;}
+}
```
Comment (chris-pearce) [inline utilities/_u-table.scss:85]: Need 2 blank lines between these type of comments:

```
/**
 * Modifier: border.
 */
```

File: utilities/_u-text.scss (hunk 1 above)
Comment (chris-pearce) [inline utilities/_u-text.scss]: Wrong class selector: `u-table`.

File: utilities/_u-text.scss [hunk 4]
```diff
@@ -19,36 +27,55 @@
...
 %u-text-size-base,
 .u-text-size-base {@include font-size($font-size);}
 
+@include generate-at-breakpoints('.u-text-size-base', $text-breakpoints) {@include font-size($font-size);}
+
 // Inherit
 %u-text-size-inherit,
 .u-text-size-inherit {font-size: inherit;}
 
+@include generate-at-breakpoints('.u-text-size-inherit', $text-breakpoints) {font-size: inherit;}
```
Comment (chris-pearce) [inline utilities/_u-text.scss:100]: Need 2 blank lines between these type of comments:

```
/**
 * Modifier: border.
 */
```

File: utilities/_u-text.scss [hunk 5]
```diff
@@ -112,18 +163,28 @@
 %u-text-capitalise-uppercase,
 .u-text-capitalise-uppercase {text-transform: uppercase;}
 
+@include generate-at-breakpoints('.u-text-capitalise-uppercase', $text-breakpoints) {text-transform: uppercase;}
+
 %u-text-capitalise-lowercase,
 .u-text-capitalise-lowercase {text-transform: lowercase;}
 
+@include generate-at-breakpoints('.u-text-capitalise-uppercase', $text-breakpoints) {text-transform: uppercase;}
```
Comment (chris-pearce) [inline utilities/_u-text.scss]: Class selector is wrong and CSS property value.

File: utilities/_u-text.scss [hunk 6]
```diff
@@ -174,6 +249,13 @@
   word-wrap: normal; // [2]
 }
 
+@include generate-at-breakpoints('.u-text-truncate', $text-breakpoints) {
+  max-width: 100%; // [1]
```
Comment (chris-pearce) [inline utilities/_u-text.scss]: No need to include the numbers again e.g. `[1]`

File: utilities/_u-text.scss [hunk 7]
```diff
@@ -174,6 +249,13 @@
...
+  overflow: hidden;
+  text-overflow: ellipsis;
+  white-space: nowrap;
+  word-wrap: normal; // [2]
+}
```
Comment (chris-pearce) [inline utilities/_u-text.scss:336]: 2 blank lines.

File: utilities/_u-toggle-visibility.scss [hunk 8]
```diff
@@ -30,6 +34,15 @@ $toggle-visibility-state-class: is-visible !default;
   .no-js & {display: block;}
 }
 
+@include generate-at-breakpoints('.u-toggle-visibility', $toggle-visibility-breakpoints) {
```
Comment (chris-pearce) [inline utilities/_u-toggle-visibility.scss]: This under 80 chars?

File: utilities/_u-toggle-visibility.scss [hunk 9]
```diff
@@ -42,6 +55,11 @@ $toggle-visibility-state-class: is-visible !default;
   .no-js & {display: inline;}
 }
 
+@include generate-at-breakpoints('.u-toggle-visibility--inline', $toggle-visibility-breakpoints) {
```
Comment (chris-pearce) [inline utilities/_u-toggle-visibility.scss]: This under 80 chars?

File: utilities/_u-toggle-visibility.scss [hunk 10]
```diff
@@ -52,4 +70,10 @@ $toggle-visibility-state-class: is-visible !default;
   // Show via state class and for non-JavaScript
   &.#{$toggle-visibility-state-class},
   .no-js & {display: inline-block;}
+}
+
+@include generate-at-breakpoints('.u-toggle-visibility--inline-block', $toggle-visibility-breakpoints) {
```
Comment (chris-pearce) [inline utilities/_u-toggle-visibility.scss]: This under 80 chars?

File: utilities/_u-toggle-visibility.scss [hunk 11]
```diff
@@ -42,6 +55,11 @@ $toggle-visibility-state-class: is-visible !default;
   .no-js & {display: inline;}
 }
 
+@include generate-at-breakpoints('.u-toggle-visibility--inline', $toggle-visibility-breakpoints) {
+  // Show via state class and for non-JavaScript
+  &.#{$toggle-visibility-state-class},
+  .no-js & {display: inline;}
+}
```
Comment (chris-pearce) [inline utilities/_u-toggle-visibility.scss:62]: 2 blank lines

File: utilities/_u-momentum-scrolling.scss (hunk 2 above)
Comment (AlecRaeside) [inline utilities/_u-momentum-scrolling.scss]: agreed

File: utilities/_u-momentum-scrolling.scss [hunk 12]
```diff
@@ -15,6 +15,14 @@
  */
 
 
+/**
+ * Settings.
+ */
+
+// Apply at these breakpoints
+$momentum-scrolling-breakpoints: $default-utility-breakpoints !default;
```
Comment (chris-pearce) [inline utilities/_u-momentum-scrolling.scss]: This isn't needed right?

File: utilities/_u-new-block-formatting-context.scss [hunk 13]
```diff
@@ -19,6 +19,15 @@
 
 
 /**
+ * Settings.
+ */
+
+// Apply at these breakpoints
+$new-block-formatting-context-breakpoints: $default-utility-breakpoints
```
Comment (chris-pearce) [inline utilities/_u-new-block-formatting-context.scss]: I'd just go with `$nbfc-breakpoints`

File: utilities/_u-new-block-formatting-context.scss [hunk 14]
```diff
@@ -32,6 +41,12 @@
   width: 10000px; // [2]
 }
 
+@include generate-at-breakpoints('.u-nbfc',
+$new-block-formatting-context-breakpoints) {
```
Comment (chris-pearce) [inline utilities/_u-new-block-formatting-context.scss]: So this looks outdented, I'll come over

File: utilities/_u-table.scss [hunk 15]
```diff
@@ -52,6 +55,19 @@ $table-striped-background-cell-colour:  $table-border-colour !default;
   td {@include to-rem(padding, $table-cell-padding-base);}
 }
 
+@include generate-at-breakpoints('.u-table', $table-breakpoints) {
+  // Stretch to full width of it's parent
```
Comment (chris-pearce) [inline utilities/_u-table.scss]: No need to repeat the comments

File: utilities/_u-text.scss [hunk 16]
```diff
@@ -19,36 +27,55 @@
 %u-text-size-small,
 .u-text-size-small {@include font-size($font-size-small);}
 
+@include generate-at-breakpoints('.u-table', $text-breakpoints) {@include font-size($font-size-small);}
+
 %u-text-size-x-small,
 .u-text-size-x-small {@include font-size($font-size-x-small);}
 
+@include generate-at-breakpoints('.u-text-size-x-small', $text-breakpoints) {@include font-size($font-size-x-small);}
```
Comment (chris-pearce) [inline utilities/_u-text.scss]: Needs to wrap at 80 chars

File: utilities/_u-toggle-visibility.scss [hunk 17]
```diff
@@ -30,6 +34,15 @@ $toggle-visibility-state-class: is-visible !default;
   .no-js & {display: block;}
 }
 
+@include generate-at-breakpoints('.u-toggle-visibility', $toggle-visibility-breakpoints) {
+  // Hide by default
```
Comment (chris-pearce) [inline utilities/_u-toggle-visibility.scss]: No need to repeat the comments

File: utilities/_u-table.scss (hunk 15 above)
Comment (chris-pearce) [inline utilities/_u-table.scss]: No need to repeat comments :)

File: utilities/_u-table.scss [hunk 18]
```diff
@@ -73,13 +94,24 @@ $table-striped-background-cell-colour:  $table-border-colour !default;
...
   }
 }
 
+@include generate-at-breakpoints('.u-table--border', $table-breakpoints) {
+  th,
+  td {
+    @include to-rem(border, $table-border-thickness $table-border-style
+      $table-border-colour);
+
+    // No borders for empty cells
```
Comment (chris-pearce) [inline utilities/_u-table.scss]: No need to repeat comments

File: utilities/_u-table.scss [hunk 19]
```diff
@@ -90,6 +122,8 @@ $table-striped-background-cell-colour:  $table-border-colour !default;
 %u-table--fixed,
 .u-table--fixed {table-layout: fixed;}
 
+@include generate-at-breakpoints('.u-table--fixed', $table-breakpoints)
+{table-layout: fixed;}
```
Comment (chris-pearce) [inline utilities/_u-table.scss]: Weird, I'll come over

File: utilities/_u-text.scss (hunk 16 above)
Comment (chris-pearce) [inline utilities/_u-text.scss]: Need to wrap loads in this file

File: utilities/_u-toggle-visibility.scss [hunk 20]
```diff
@@ -30,6 +34,16 @@ $toggle-visibility-state-class: is-visible !default;
   .no-js & {display: block;}
 }
 
+@include generate-at-breakpoints('.u-toggle-visibility',
+  $toggle-visibility-breakpoints) {
+  // Hide by default
```
Comment (chris-pearce) [inline utilities/_u-toggle-visibility.scss]: No need to repeat comments

**Comment by chris-pearce** [comment]
LGTM

**Pull request merged**